apiserver/getcachedvhdimages.go

```diff
@@ -0,0 +1,41 @@
+package apiserver
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	agent "github.com/Azure/agentbaker/pkg/agent"
+)
+
+const (
+	// RoutePathGetCachedVHDImages the route path to get cached vhd images.
+	RoutePathGetCachedVHDImages string = "/getcachedvhdimages"
+)
+
+// GetCachedVHDImages endpoint for getting cached VHD images.
+func (api *APIServer) GetCachedVHDImages(w http.ResponseWriter, r *http.Request) {
```

Comment:
I'd be curious to see on the RP side how this would get used

im also wondering if we'd ever run into problems where we cache certain things on certain SKUs but not others - in that case just returning contents of the manifest wouldn't quite work